src/dev/i18n/extractors/html.js

```diff
@@ -53,7 +54,7 @@ function parseFilterObjectExpression(expression, messageId) {
 
   try {
     // parse an object expression instead of block statement
-    ast = parse(`+${expression}`);
+    ast = parse(`+${expression}`.replace(LINEBREAK_REGEX, ' '));
```

Comment:
__nit:__ please move `parse(expression.replace(LINEBREAK_REGEX, '  '))` to a separate `parseExpression` function that will be used in all 3 cases. I think it's even fine if you move ` if (error instanceof SyntaxError) {....}` there as well and remove `try/catch` from all 3 cases entirely. Generic `Couldn't parse angular i18n expression:\n${errorWithContext}` error message for all 3 cases should be enough to figure out the reason of failure.